app/win/record.js

```diff
@@ -0,0 +1,37 @@
+// Application state fallback upon unexpected quit
+const {app} = require('electron');
+const Config = require('electron-config');
+const wins = require('./windows');
+
+// local storage
+const rec = new Config();
+const recordInterval = 2000;
+
+module.exports.save = function (windows) {
+  setInterval(() => {
+    const states = [];
+    windows.forEach(win => {
+      win.record(state => {
+        states.push(state);
+      });
+    });
+    rec.set('reccords', states);
+  }, recordInterval);
+};
+
+module.exports.load = function () {
+  const reccords = rec.get('reccords');
+  if (reccords.length > 0) {
```

Comment:
You need a `reccords && reccords.length > 0` here – without it, you won't be able to run Hyper for the first time:

```
▲ hyper at ppot-hyper_record ✔ npm run app

> @ app /Users/matheus/dev/zeit/hyper
> electron app

plugins have changed / not init, scheduling plugins installation
running in dev mode
electron will open file:///Users/matheus/dev/zeit/hyper/app/index.html
Error while loading devtools extensions TypeError: Cannot read property 'length' of undefined
    at Object.module.exports.load (/Users/matheus/dev/zeit/hyper/app/win/record.js:24:15)
    at installDevExtensions.then (/Users/matheus/dev/zeit/hyper/app/index.js:160:10)
```